assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Idea.js

```diff
@@ -203,13 +248,14 @@ export default function Idea( props ) {
 						) }
 
 						{ buttons.includes( IDEA_HUB_BUTTON_CREATE ) && (
-							<Button
+							<IdeaActivityButton
+								activity={ IDEA_HUB_BUTTON_CREATE }
 								className="googlesitekit-idea-hub__actions--create"
-								onClick={ handleCreate }
-								disabled={
-									activity === IDEA_HUB_ACTIVITY_IS_PROCESSING
+								inProgress={
+									activity ===
+									IDEA_HUB_ACTIVITY_CREATING_DRAFT
 								}
-								icon={ <CreateIcon /> }
+								onClick={ handleCreate }
```

Comment:
@aaemnnosttv I think adding different activity button might result in having some unnecessary extra files. I updated the PR so that `IdeaActivityButton` is a bit more smart, getting most of its information from maps. We can even have the handlers defined in a map and then call them accordingly, but I think we'll just moving most of the code from `Idea` to `IdeaActivityButton` which will result in the same issue, i.e a large component.
Let me know what you think, happy to go with your decision 😁 even though am not a big fan of having those individual button components.

Reply:
I was thinking about this initially too. This works well for now I think although I'm still somewhat thinking keeping the components separate might be better. I don't feel very strongly about it though so we can always revisit if needed 👍